pkg/metricsadapter/controller.go

```diff
@@ -62,7 +63,7 @@ func NewMetricsController(restConfig *rest.Config, factory informerfactory.Share
 		InformerFactory:       factory,
 		ClusterLister:         clusterLister,
 		MultiClusterDiscovery: multiclient.NewMultiClusterDiscoveryClient(clusterLister, kubeFactory),
-		InformerManager:       genericmanager.GetInstance(),
+		InformerManager:       typedmanager.GetInstance(),
```

Comment:
`karmada-metrics-adapter` can have its transformer functions, and doesn't have to share this one which is used by `karmada-controller-manager`.

@chaosi-zju Can you help to confirm if these fields enough for `karmada-metrics-adapter`?
Nodes: https://github.com/karmada-io/karmada/blob/adef1e59748e1e1d31cb75fffe406b5dd69a66d7/pkg/util/fedinformer/transform.go#L67-L75
Pods: https://github.com/karmada-io/karmada/blob/adef1e59748e1e1d31cb75fffe406b5dd69a66d7/pkg/util/fedinformer/transform.go#L100-L118

Reply:
Updated.